ted2zim/scraper.py

```diff
@@ -299,16 +313,27 @@ def extract_videos_on_page(self, page_html, video_allowance):
         # link to the representative TED talk. It turns this relative link to
         # an absolute link and calls extract_video_info for them
         soup = BeautifulSoup(page_html, features="html.parser")
-        videos = soup.select("div.row div.media__image a")
+        video_links = soup.select("div.row div.media__image a")
         nb_extracted = 0
-        logger.debug(f"{str(len(videos))} video(s) found on current page")
-        for video in videos:
-            url = urljoin(self.talks_base_url, video["href"])
-            if self.extract_video_info(url):
-                nb_extracted += 1
-                if nb_extracted == video_allowance:
-                    break
-            logger.debug(f"Seen {video['href']}")
+        logger.debug(f"{str(len(video_links))} video(s) found on current page")
+        for video_link in video_links:
+            url = urllib.parse.urljoin(self.talks_base_url, video_link["href"])
+            if not any(
```

Comment:
`not any()` feels unnatural and forces to double-think it :)

```py
if not [video.get("tedpath", None) == video_link["href"] for video in self.videos]:
```

Reply:
Changed it